main.py

```diff
@@ -1,20 +1,68 @@
 from derivations import *
 from structures import *
+import os
 
-def main():
-    lexicon = Lexicon()
+filename = "lexicon.xml"
+
+def print_grammars():
+    global filename
+    full_path = os.path.abspath("data")
+    print(full_path)
+    filenames = [f for f in os.listdir(full_path) if os.path.isfile(os.path.join(full_path, f))]
```

Comment:
Consideraría la posibilidad de filtrar por extensión de archivo. Pensá que si mañana querés agregarun `README.md`a la carpeta `data` para que futuros usuarios sepan cómo construir una gramática, ese README aparecería listado porque es un file. Si lo filtrás por extensión, reducís ese riesgo.

```suggestion
    filenames = [f for f in os.listdir(full_path) if f.endswith('.xml')]
```